pipenv/utils.py

```diff
@@ -264,6 +267,22 @@
 ]
 
 
+def get_requirement(dep):
+    """Resolve a requirement no matter where it points"""
+    path = None
+    if (is_file(dep) and isinstance(dep, six.string_types) and
+            not any(dep.startswith(uri_prefix) for uri_prefix in FILE_LIST)):
+        dep_path = Path(dep)
+        if dep_path.exists() and not dep_path.is_absolute():
+            path = get_converted_relative_path('{}'.format(dep))
```

Comment:
Why can't we pass `dep` here? Haven't we already asserted it's `str`-like?

Reply:
On second pass, doesn't this exacerbate the problem with local packages vs pypi packages?

If I have a directory named `ansible` in my package, and specify that I was to install `ansible`, I would say the correct behaviour is to install from pypi. If I want to install from my local directory, it's up to the me to dictate that (`./ansible`).

With this approach, it looks like a pypi install will never be possible.

Reply:
I think we can pass `dep`, I wrote this section in kind of a hurry which will explain the next piece of information.

I was looking at this briefly yesterday before I stopped working on this, but I think you're right about the behavior of `./ansible`.  That is to say, with this current code, `pipenv install ansible` will   accurately install ansible from pypi actually, but it won't write it to the lockfile.  I actually think I have a different approach to handle this in general though and I'm curious about whether it will work.  I will update the PR shortly.